emit-ch4plume-v1/src/index.js

```diff
@@ -25,525 +25,772 @@ const markerProps = new Object();
-        this.container.parentNode.removeChild(this.container);
-        this.map = undefined;
-    }
+      } catch (error) {
+        console.error(error);
+      }
+    });
+  }
+  onAdd(map) {
+    this.map = map;
+    this.container = document.createElement("div");
+    this.container.className = "mapboxgl-ctrl mapboxgl-ctrl-group";
+    this.container.addEventListener("contextmenu", (e) => e.preventDefault());
+    this.container.addEventListener("click", (e) => this.onClick());
+    this.container.innerHTML =
+      '<div class="tools-box">' +
+      "<button>" +
+      '<span id="layer-eye" class="mapboxgl-ctrl-icon btn fa fa-eye" aria-hidden="true" title="Show/Hide layers"></span>' +
+      "</button>" +
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
+
+class MeasureDistance {
+  onClick() {
+    $("#measure-icon").toggleClass("measure-icon-clicked");
+    measureToggled = !measureToggled;
+    $("#display_props").toggleClass("hidden");
+  }
+  onAdd(map) {
+    this.map = map;
+    this.container = document.createElement("div");
+    this.container.className = "mapboxgl-ctrl mapboxgl-ctrl-group";
+    this.container.addEventListener("contextmenu", (e) => e.preventDefault());
+    this.container.addEventListener("click", (e) => this.onClick());
+    this.container.innerHTML =
+      '<div id="measure-icon-main">' +
+      "<button>" +
+      '<span id="measure-icon" class="mapboxgl-ctrl-icon" aria-hidden="true" title="Measure Tool">' +
+      '<i class="fa-solid fa-ruler-horizontal"></i>' +
+      "</span>" +
+      "</button>" +
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
+class ClearDistancePoints {
+  onClick() {
+    distancePoints.features.splice(0, distancePoints.features.length);
+     measureLine.features.splice(0, measureLine.features.length);
+     distanceLabel.features.splice(0, distanceLabel.features.length);
+     map.getSource("distanceLabel").setData(distanceLabel);
+     map.getSource("measureLine").setData(distanceLabel);
+    map.getSource("distancePoints").setData(distancePoints);
+    $("#clear-icon-main").removeClass("clicked");
+  }
+  onAdd(map) {
+    this.map = map;
+    this.container = document.createElement("div");
+    this.container.className = "mapboxgl-ctrl mapboxgl-ctrl-group";
+    this.container.addEventListener("contextmenu", (e) => e.preventDefault());
+    this.container.addEventListener("click", (e) => this.onClick());
+    this.container.innerHTML =
+      '<div id="clear-icon-main">' +
+      "<button>" +
+      '<span id="clear-icon" class="mapboxgl-ctrl-icon" aria-hidden="true" title="Measure Tool">' +
+      '<i class="fa-solid fa-eraser"></i>' +
+      "</span>" +
+      "</button>" +
+      "</div>";
+    return this.container;
+  }
+  onRemove() {
```

Comment:
This seems to be reused in many other control classes. could create an abstract class with this `onRemove()` reused definition. Inheriting that would remove the need for this one in  `ClearDistancePoints` control class. More DRY the better.